### PR TITLE
Nav2_tasks readme

### PR DESCRIPTION
**Pull request by mjeronimo**
Currently, there are many empty README.md files across the various directories. These are placeholders that we plan to populate during October as we work towards a functional stack. Each will contain background information for the particular package and a description of its contents. This particular PR is a first cut at the README file for the nav2_tasks directory. I created the diagram using LibreOffice Draw and exported it to a .svg file and checked in both to a doc subdirectory (is this how we want to do it?).

### CODE REVIEW

File: nav2_tasks/README.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    }
+  }  
+}
+```
+
+## Implementation
+
+The implementation of TaskClient and TaskServer is simplistic as the intention is to rely upon a ROS2 ActionLib (or equivalent) when it is available. 
+
+The current implementation uses four topics, one each for **Command**, **Cancel**, **Result**, and **Status** messages. The Command and Result messages are provided by the user of the classes, while the Result and Status messages are internal to the implementation. The Status message communicates whether a task has succeeded, failed, or has been canceled, while the Result message is used to return the result of the command, if successful. To keep things simple, the task library does not implement a feedback mechanism as the ROS1 ActionLib does.
```
Comment (bpwilcox) [inline nav2_tasks/README.md]: Do you mean "The Command and _Cancel_ messages are provided..."? I think you forgot to mention the Cancel topic.

File: nav2_tasks/README.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+* [TaskServer](include/nav2_tasks/task_server.hpp) and [TaskClient](include/nav2_tasks/task_client.hpp) templates which are used as base classes to implement task servers and their associated task clients
+* A few derived classes, such as [ComputePathToPoseTask](include/nav2_tasks/compute_path_to_pose_task.hpp), that define specific task clients and servers
+
+For convenience the nav_tasks library also provides:
+* A [ServiceClient](include/nav2_tasks/service_client.hpp) template used to define clients for ROS2 services
+* A couple specific service client classes, such as [MapServiceClient](include/nav2_tasks/map_service_client.hpp) that use the ServiceClient template
+
+## Overview
+
+The Navigation2 architecture implements a *task hiearchy* where commands are sent to modules which are organized in a hierarchy. Each module implements a task and can utilize sub-tasks, which are themselves modules that implement tasks.
```
Comment (bpwilcox) [inline nav2_tasks/README.md]: mispelled "hierarchy"

**Comment by mjeronimo** [comment]
@crdelsey Actually, I'm deleting the pyc file. I noticed that it was currently in master and getting rid of it. It must've snuck in with another commit.

**Comment by mjeronimo** [comment]
@bpwilcox Please have a look at the updated text discussing the messages.

**Comment by crdelsey** [comment]
I thought we had just merged a change deleting the pyc file so I thought this must be adding it back. Sorry.

**Pull request merged**